File: mobile/ios/Runner/PasskeyHandler.swift
import Foundation
import Flutter

/// Stub handler for the `healthvault.security/passkey` MethodChannel.
///
/// TODO(passkey-backend): replace with a real AuthenticationServices
/// integration (ASAuthorizationController + ASAuthorizationPlatform
/// PublicKeyCredentialProvider) once the Go API gains WebAuthn endpoints.
///
/// The current impl returns unavailable / unimplemented so the Dart side
/// can show a "coming soon" UI instead of crashing.
final class PasskeyHandler: NSObject {

    static let channelName = "healthvault.security/passkey"

    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "support":
            result(false)
        case "register", "authenticate":
            result(FlutterError(
                code: "not_implemented",
                message: "Passkey backend not yet available",
                details: nil
            ))
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: mobile/android/app/src/main/kotlin/de/kiefer_networks/healthapp/PasskeyHandler.kt
package de.kiefer_networks.healthapp

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

/**
 * Stub handler for the `healthvault.security/passkey` MethodChannel.
 *
 * TODO(passkey-backend): replace this stub with a real implementation
 * using androidx.credentials CredentialManager once the Go API gains
 * WebAuthn endpoints that can issue challenges and verify attestations.
 *
 * Expected future methods:
 *   support()     -> Bool
 *   register(...) -> ByteArray (attestation)
 *   authenticate(...) -> ByteArray (assertion)
 *
 * The current impl returns unavailable / unimplemented so the Dart side
 * can show a "coming soon" UI instead of crashing.
 */
class PasskeyHandler : MethodChannel.MethodCallHandler {

    companion object {
        const val CHANNEL = "healthvault.security/passkey"
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "support" -> result.success(false)
            "register", "authenticate" -> {
                result.error(
                    "not_implemented",
                    "Passkey backend not yet available",
                    null,
                )
            }
            else -> result.notImplemented()
        }
    }
}
